### ISSUE TITLE
SelectedValue was not maintained when changing the Text property in WINUI ComboBox

### ISSUE THREAD

**Issue by SreemonPremkumarMuthukrishnan** [microsoft/microsoft-ui-xaml#7181]
### Describe the bug

In WPF, when the SelectedValue and Text property is assigned to a value, the SelectedValue property is maintained by the Text property value and it is shows on the UI.
 
WPF Reference image:

![GIT_WPF_AfterReachingTextValue](https://user-images.githubusercontent.com/104507189/171622034-a3a1c61c-0f3e-4bc1-a027-c3ae53e4a84a.png)

For the same scenario, in WINUI, SelectedValue is not maintained and does not show on the UI. 

WINUI Reference image:

![GIT_WINUI_AfterReachingTextValue](https://user-images.githubusercontent.com/104507189/171622180-579c12a2-ce81-4795-a947-62ea16a2e889.png)

Can anyone explain this behavior?

### Steps to reproduce the bug

1. Create ComboBox in both WINUI and WPF.
2. Assign any values from the ComboBox  to SelectedValue and Text property.

### Windows app type

- [ ] UWP
- [ ] Win32

**Comment by kmgallahan** [comment]
Are you using an editable combobox?

**Comment by SreemonPremkumarMuthukrishnan** [comment]
@kmgallahan Yes, its an editable combobox. can you tell me what is the behavior?

**Comment by RossCoeH** [comment]
Looks as expected to me.  It looks like you assign a selected value, then the next line assigns the text property, which will change the value of the combobox. You will not necessarily have a selected value because the text may not match an existing item in the list.  Changing the text value is not equivalent to editing the item in the list. Perhaps you should explain what the expected behavior should be..

**Comment by SreemonPremkumarMuthukrishnan** [comment]
@RossCoeH When you assign a value to the Text property in WPF, the SelectedValue property is maintained, but not in WINUI. Consider the following scenario: we type a value in the ComboBox that isn't in the ComboBox list; the written value isn't maintained in the WINUI selection box, but it is maintained in WPF. Is WINUI behavior is correct?
